### PR TITLE
Added filter, fixed a bug with null values

### PR DESCRIPTION
**Pull request by tavurth**
- [x] Filtered for only TestItem trait types
- [x] Used Option as change values can sometimes be null

### CODE REVIEW

File: examples/changelog.rs [hunk 1]
```diff
@@ -20,13 +20,21 @@ use reql::{Client, Document, Run};
  *
  * setup a "test" database, with a "test" table, and run:
  *
- * r.db('test').table('test').insert({ test: 1 })
+ * // Insert an item
+ * r.db('test').table('test').insert({ test: 1 });
```
Comment (rushmorem) [inline examples/changelog.rs]: Rust doesn't support anonymous documents like `{ test: 1 }` use the [json macro](https://docs.rs/serde_json/1.0.5/serde_json/macro.json.html) instead. Also note that single quotes are used for `char`. So this query should be:-

```rust
r.db("test").table("test").insert(json!({ "test": 1 }));
```

File: examples/changelog.rs [hunk 2]
```diff
@@ -20,13 +20,21 @@ use reql::{Client, Document, Run};
...
+ *
+ * // Give the first item a random number
+ * r.db('test').table('test').nth(0).update({ test: r.random(0, 100) }, { nonAtomic: true });
```
Comment (rushmorem) [inline examples/changelog.rs]: `update` needs to use the `args` macro here.

File: examples/changelog.rs (hunk 1 above)
Comment (tavurth) [reply examples/changelog.rs]: These actions are actually javascript, as they were intended to be input in the rethinkdb admin panel's data-explorer.

If you think it's more useful, I can add them at the end of the file as actual rust commands, which execute when the file is run?

**Comment by rushmorem** [comment]
I forgot to mention this in the previous pull request but you don't need to redefine the `Change` struct. It's [already provided](https://docs.rs/reql-types/0.0.1/reql_types/struct.Change.html) by the [reql-types crate](https://docs.rs/reql-types). Here is an example of its use https://github.com/rethinkdb-rs/reql/blob/918b65ee58a13b62c877830b560b97b1ecb0fc18/examples/server-status.rs#L40

File: examples/changelog.rs (hunk 1 above)
Comment (rushmorem) [reply examples/changelog.rs]: Oop! My bad. Never mind, then. I had not read your instructions just before those commands.

Comment (rushmorem) [reply examples/changelog.rs]: > Edit: Perhaps we can add the equivalent commands below, so as to show how to run these inserts in another thread, I'll do that just now.

That's not really necessary but if you are up for it, then knock yourself out :)

**Comment by rushmorem** [comment]
I have dismissed the previous review. It's upto you if you still want to pursue it. However, you might want to consider using `Change` from the `reql-types` crate so our users don't think they have to redefine it in their own code.

**Comment by tavurth** [comment]
So the issue I'm having now is that the `reql` change type returns an Option<String>, which I'm having difficulty matching.

I'm looking at something similar to the following:
```rust
let action = match change.result_type {
    Some(action) => Ok(action.as_ref()),
    None => Err("Invalid change type")
};

// Extract the change type
print!("{:?} action received\n\t=> ", action);

// Match the change type
match action {
    "add" => println!("{:?}", change.new_val),
    "remove" => println!("{:?}", change.old_val),
    "change" => println!("from {:?} to {:?}", change.old_val, change.new_val),

    _ => println!("{:?}", change)
}
```

But I'm not sure yet how to match that Result to a &str. Perhaps I'm going about this in a way which isn't rusty?

**Comment by tavurth** [comment]
So I've updated with the version which uses the `reql:Change` struct, although it's not quite as clean now. I guess it's better to be making sure we've got a change type here?

**Comment by rushmorem** [comment]
The problem is that your action is not a `&str` but `Result<&str, &str>` so in your match statements you will need to use `Ok("add")`, `Ok("remove")` etc.

You could simply use an if let, eg:-

```rust
if let Some(action) = change.result_type {
    // Match the change type 
    match action.as_str() {
        "add" => println!("{:?}", change.new_val),
        "remove" => println!("{:?}", change.old_val),
        "change" => println!("from {:?} to {:?}", change.old_val, change.new_val),

        _ => println!("{:?}", change)
    }
} else {
    println!("Invalid change type");
}
```

**Comment by tavurth** [comment]
Thanks, that looks a lot more like what I'd expect from rust. I've made the changes.

**Comment by rushmorem** [comment]
No problem. Thanks again for the PR!

**Pull request merged**